http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -168,6 +177,14 @@ private void addProxyConfig(HttpClientBuilder builder,
                 builder.setDefaultCredentialsProvider(ApacheUtils.newProxyCredentialsProvider(proxyConfiguration));
             }
         }
+
+        if (configuration.credentialsProvider != null) {
+            if (isAuthenticatedProxy(proxyConfiguration)) {
+                log.warn(() -> "Proxy configuration username/password is ignored whe using a custom Credential Provider");
+            }
+
+            builder.setDefaultCredentialsProvider(configuration.credentialsProvider);
```

Comment:
Can you refactor a bit here to combine the setDefaultCredentialsProvider here and in the previous block?

Reply:
Done